homeassistant/components/xbox/binary_sensor.py

```diff
@@ -0,0 +1,143 @@
+        if self.attribute == "online":
+            return self.data.gamertag
+
+        attr_name = " ".join([part.title() for part in self.attribute.split("_")])
+        return f"{self.data.gamertag} {attr_name}"
+
+    @property
+    def entity_picture(self) -> str:
+        """Return the gamer pic."""
+        return self.data.display_pic.replace("&mode=Padding", "")
+
+    @property
+    def is_on(self) -> bool:
+        """Return the status of the requested attribute."""
+        if not self.coordinator.last_update_success:
+            return False
+        return getattr(self.data, self.attribute)
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, str]]:
+        """Return friend attributes."""
+        if self.attribute != "online":
+            return None
+
+        return {
+            "status": self.data.presence_text,
+            "gamer_score": self.data.gamer_score,
+            "account_tier": self.data.account_tier,
+            "tenure": self.data.tenure,
+        }
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return self.attribute == "online"
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            "identifiers": {(DOMAIN, "xbox_live")},
+            "name": "Xbox Live",
+            "manufacturer": "Microsoft",
+            "model": "Xbox Live",
+            "entry_type": "service",
+        }
+
+
+@callback
+def async_update_friends(
+    coordinator: XboxUpdateCoordinator,
+    current: Dict[str, List[XboxBinarySensorEntity]],
+    async_add_entities,
+) -> None:
+    """Update friends."""
+    new_ids = set(coordinator.data.presence)
+    current_ids = set(current)
+
+    # Process new favorites, add them to Home Assistant
+    new_entities = []
+    for xuid in new_ids - current_ids:
+        current[xuid] = [
+            XboxBinarySensorEntity(coordinator, xuid, attribute)
+            for attribute in PRESENCE_ATTRIBUTES
+        ]
+        new_entities = new_entities + current[xuid]
+
+    if new_entities:
+        async_add_entities(new_entities)
+
+    # Process deleted favorites, remove them from Home Assistant
+    for xuid in current_ids - new_ids:
+        coordinator.hass.async_create_task(
+            async_remove_entities(xuid, coordinator, current)
+        )
+
+
+async def async_remove_entities(
+    xuid: str,
+    coordinator: XboxUpdateCoordinator,
+    current: Dict[str, XboxBinarySensorEntity],
+) -> None:
+    """Remove friend sensors from Home Assistant."""
+    entities = current[xuid]
+    for entity in entities:
+        await entity.async_remove()
+        registry = await async_get_entity_registry(coordinator.hass)
+        if entity.entity_id in registry.entities:
+            registry.async_remove(entity.entity_id)
```

Comment:
Don't forget to remove the device entry as well

Reply:
This is here to dynamically update sensors based on what friends you have "favorited" on your profile. We are only removing sensors that have been removed as favorites on your friend's list. The device entry doesn't ever get removed unless the entire integration is removed.